### ISSUE TITLE
cannot determine current apt mirror

### ISSUE THREAD

**Issue by wozeparrot** [vegardit/fast-apt-mirror.sh#2]
Not sure what happened or what could have even caused this, it might just be actions being weird, seems to be pretty recent.

Seen here: https://github.com/tinygrad/tinygrad/actions/runs/5881949748/job/15951433949?pr=1558
here: https://github.com/tinygrad/tinygrad/actions/runs/5881949748/job/15951433775?pr=1558
here: https://github.com/tinygrad/tinygrad/actions/runs/5881931452/job/15951377952?pr=1537
here: https://github.com/tinygrad/tinygrad/actions/runs/5875504570/job/15931888916?pr=1076
here: https://github.com/tinygrad/tinygrad/actions/runs/5875504570/job/15931889162?pr=1076

```
  APT::Acquire::Retries "3";
  + curl -sSfL --retry 3 --max-time 6 https://raw.githubusercontent.com/vegardit/fast-apt-mirror.sh/v1/fast-apt-mirror.sh -o /home/runner/fast-apt-mirror.sh
  Current mirror: unknown
  Randomly selecting 20 mirrors...done
  Checking health status of 20 mirrors using '/dists/focal-security/Contents-amd64.gz'....................done
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) http://mirror.umd.edu/ubuntu/
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) http://mirror.umd.edu/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://mirrors.rit.edu/ubuntu/
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) http://mirror.mrjester.net/ubuntu/archive/
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) http://mirror.mrjester.net/ubuntu/archive/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://mirror.steadfastnet.com/ubuntu/
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) http://plug-mirror.rcac.purdue.edu/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://mirror.pit.teraswitch.com/ubuntu/
   -> outdated   (last modified: 2023-08-01 05:51:46 UTC) https://mirror.mia.velocihost.net/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://repos.forethought.net/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://www.gtlib.gatech.edu/pub/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://mirror.us-sc.kamatera.com/ubuntu/
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) https://mirrors.wikimedia.org/ubuntu/
   -> outdated   (last modified: 2023-08-15 08:47:44 UTC) http://mirror.uoregon.edu/ubuntu/
   ->                         n/a                          https://mirror.enzu.com/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) http://mirrors.syringanetworks.net/ubuntu-archive/
   ->                         n/a                          http://ftp.ussg.iu.edu/linux/ubuntu/
   -> outdated   (last modified: 2020-08-19 04:19:51 UTC) http://mirrors.arpnetworks.com/Ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) https://mirror.lstn.net/ubuntu/
   -> UP-TO-DATE (last modified: 2023-08-16 06:07:16 UTC) https://repo.ialab.dsu.edu/ubuntu/
   => 9 mirrors are reachable and up-to-date
  Speed testing 9 of the available 9 mirrors (sample download size: 1024KB).........done
   -> http://mirrors.syringanetworks.net/ubuntu-archive/ (782KB/s)
   -> http://repos.forethought.net/ubuntu/ (1.1MB/s)
   -> http://mirror.us-sc.kamatera.com/ubuntu/ (2.0MB/s)
   -> https://repo.ialab.dsu.edu/ubuntu/ (2.7MB/s)
   -> https://mirror.lstn.net/ubuntu/ (2.8MB/s)
   -> http://mirror.pit.teraswitch.com/ubuntu/ (3.5MB/s)
   -> http://mirror.steadfastnet.com/ubuntu/ (5.2MB/s)
   -> http://www.gtlib.gatech.edu/pub/ubuntu/ (5.7MB/s)
  Current mirror: unknown
  ERROR: Cannot set APT mirror: cannot determine current APT mirror.
 ```

**Comment by wozeparrot** [comment]
Oh a new runner image is out, and the sources are different now: https://github.com/actions/runner-images/commit/7fa63e2b95b104d0a626378e744c575937883c99

**Comment by sebthom** [comment]
Hm, I cannot reproduce this. I just reran the test job https://github.com/vegardit/fast-apt-mirror.sh/actions/runs/5882318045/job/15952602786 which tests it using these images:
- ubuntu-latest
- ubuntu-22.04
- ubuntu-20.04

Which images are the failing builds using?

**Comment by wozeparrot** [comment]
We're running `ubuntu-latest`, it seems they are doing a slow rollout of the new image version. I see the failures on `20230814.1.0` but your test jobs are still on `20230806.1.0`.

**Comment by sebthom** [comment]
I tried it in several repos, but don't get the new release. `20230814.1.0` is still flagged as a pre-release at https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230814.1

We need to update the `get_current_mirror` function at https://github.com/vegardit/fast-apt-mirror.sh/blob/v1/fast-apt-mirror.sh#L138 but without having access to a shell on the new image I don't know how yet.

**Comment by wozeparrot** [comment]
Yea, that makes sense, wonder why we are already on the new image tho.

**Comment by sebthom** [comment]
I got "lucky" now. Some of my actions use the new image.

**Comment by sebthom** [comment]
Fixed via de344d53e26d041a7ac293ef6c69f8106bd8433a
Thanks for reporting!
